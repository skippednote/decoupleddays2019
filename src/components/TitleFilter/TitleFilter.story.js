import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import TitleFilter from ".";

storiesOf("Title Filter", module).add("default", () => (
  <TitleFilter filter={""} setFilter={action()} />
));
