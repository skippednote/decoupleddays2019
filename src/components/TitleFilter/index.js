import React from "react";

function TitleFilter({ filter, setFilter }) {
  return (
    <div className="filter-title">
      <input
        data-testid="filter-input"
        type="text"
        placeholder="Filter by title"
        className="filter-title"
        onChange={e => setFilter({ ...filter, input: e.target.value })}
      />
    </div>
  );
}

export default TitleFilter;
