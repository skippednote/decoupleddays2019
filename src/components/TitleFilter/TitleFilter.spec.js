import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TitleFilter from ".";

it("can render the component", () => {
  let filter = { input: "" };
  const setFilterMock = jest.fn();

  const { container } = render(
    <TitleFilter filter={filter} setFilter={setFilterMock} />
  );

  expect(container).toMatchSnapshot();
});

it("can update the filter on click", () => {
  let filter = { input: "" };
  const setFilterMock = jest.fn();

  const { getByTestId } = render(
    <TitleFilter filter={filter} setFilter={setFilterMock} />
  );

  const input = getByTestId("filter-input");
  fireEvent.input(input, {
    target: {
      value: "Testing Front-end"
    }
  });

  expect(setFilterMock).toHaveBeenCalledTimes(1);
  expect(setFilterMock).toHaveBeenCalledWith({ input: "Testing Front-end" });
});
