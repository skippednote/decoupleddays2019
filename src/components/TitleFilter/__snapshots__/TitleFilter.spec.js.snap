// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render the component 1`] = `
<div>
  <div
    class="filter-title"
  >
    <input
      class="filter-title"
      data-testid="filter-input"
      placeholder="Filter by title"
      type="text"
    />
  </div>
</div>
`;
